client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -652,6 +666,11 @@ void reconnect() {
         }
     }
 
+    private void failUnackedEvents(Throwable t) {
+        state.getAllInflightEvents().parallelStream().forEach(event -> event.getAckFuture().completeExceptionally(t));
```

Comment:
If they are completed exceptionally they should probably be removed from inflight as they cannot be subsequently be completed successfully.

Reply:
ok, I have updated the code.